app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+  has_many :posts, foreign_key: 'author_id'
+
```

Comment:
- [ ] Merely changing the models, does not change the database schema in the case. As pointed out earlier these changes need to be reflected in the database tables.  What you currently have right now in your columns is `user_id` whereas you should have `author_id`. You need to run migrations to change the column names.

  Post schema | Comment schema | Like schema
  :---:|:---:|:---:
  ![sadiq-post-user](https://user-images.githubusercontent.com/36542986/162958833-63da3c96-d09f-4d77-97ef-5d7c26205a6c.jpg) | ![sadiq-comment-user](https://user-images.githubusercontent.com/36542986/162958822-13e65a25-78ce-44ce-a4f6-66f10ea61a3b.jpg) | ![sadiq-like-user](https://user-images.githubusercontent.com/36542986/162958830-08912d86-224f-4229-bfa3-3ae3874ea2e4.jpg)

  ### SUGGESTION:
  You need to run new migrations to change the schema. You can refer to this [Stackoverflow answer](https://stackoverflow.com/a/1992045) on how to rename columns in Rails.

  For example:
  Run: `rails g migration ChangePostUserIdToAuthorId`
  
  ```
  # db/migrate/migration file
  class ChangeUserIdToAuthorId < ActiveRecord::Migration[7.0]
    def change
      rename_column :posts, :user_id, :author_id
    end
  end
  ```